### PR TITLE
MES-7577 Refactor Log in

### PR DESCRIPTION
**Pull request by Cullima**
## Description

- Fixes log in (on fresh build and test)
- Bumps various wdio packages to enable `async` migration
- Bumps appium and iOS versions

## Checklist

- [ ] PR title includes the JIRA ticket number and an understandable description
- [ ] Branch is rebased against the latest develop
- [ ] Code has been tested manually
- [ ] PR link added to JIRA ticket

### CODE REVIEW

**Review by matthew2564 — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -3,18 +3,21 @@ DVSA Automation Framework
 #### Requirements
 
 - Node version 12 or higher
```
Comment (dhDVSA) [inline README.md:5]:
```suggestion
- Node version 14 or higher
```

**Review by gabrielngdvsa — APPROVED** [review]

File: config/wdio.DES-Appium.conf.ts [hunk 2]
```diff
@@ -47,7 +47,7 @@ exports.config = {
             // app: '/Users/lee/IonicProjects/dvsa-automation-boilerplate/apps/SearchMobile.app',
             app: DesInfo.localAppPath,
             disableAnimations: true,
-            fullReset: false, // Change this to true to reinstall the application
+            fullReset: true, // Change this to true to reinstall the application
```
Comment (dhDVSA) [inline config/wdio.DES-Appium.conf.ts:50]: this should probably be set to false by default as these tests will run in ci

File: config/wdio.conf.ts [hunk 3]
```diff
@@ -220,12 +220,6 @@ export const config: WebdriverIO.Config = {
 
     before: async (capabilities: any, specs: string[], browser: any) => {
         console.log('>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> Running Before Script <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<');
```
Comment (dhDVSA) [inline config/wdio.conf.ts:222]:
```suggestion
```

File: src/pageobjects/des/login.pageobject.ts [hunk 4]
```diff
@@ -1,174 +1,176 @@
...
+  // private _msSignOutContextTitle: string = 'Sign out';
+  private _desAppContextTitle: string = 'Ionic App';
 
-  getSuperUser() {
-    const totalUsers = credentials.Environment.Dev.Super.length
-    const randomUserId = Math.floor(Math.random() * totalUsers);
-    return credentials.Environment.Dev.Super[randomUserId - 1];
+  async doesContextExist(contextTitle: string): Promise<boolean> {
+    const contexts: AppiumContext[] = await driver.getContexts() as unknown as AppiumContext[];
+    console.log(contexts);
```
Comment (dhDVSA) [inline src/pageobjects/des/login.pageobject.ts:21]:
```suggestion
    console.log(`Apium Contexts : ${contexts}`);
```

File: config/wdio.DES-Appium.conf.ts (hunk 2 above)
Comment (Cullima) [reply config/wdio.DES-Appium.conf.ts:50]: eventually but only after the already logged in scenario is handled. BS disregards this capability anyway

File: src/pageobjects/des/login.pageobject.ts [hunk 5]
```diff
@@ -1,174 +1,176 @@
...
+      timeoutMsg: `Element with selector: ${selector} was not clickable on page within 15 seconds`,
+    });
   }
 
-  checkAlreadyLoggedIn() {
-    return !this.doesContextExistByArray([this._contextTitleForPage]);
+  async clickNativeButton(element: WebdriverIO.Element): Promise<void> {
+    await this.waitForContextToExist('NATIVE_APP');
+    await driver.switchContext('NATIVE_APP');
+    await browser.pause(3000);
```
Comment (dhDVSA) [inline src/pageobjects/des/login.pageobject.ts:64]: We should probably add a waitUntil here for the context to change rather than implicitly pause for a specified time

File: src/pageobjects/des/login.pageobject.ts [hunk 6]
```diff
@@ -1,174 +1,176 @@
...
-      driver.deleteAllCookies();
-      console.log('Refresh login page');
-      driver.refresh();
-    }
+  getSuperUser() {
+    return credentials.Environment.Dev.Super[0];
   }
 
-  getRandomUserType( userType:string) {
+  getRandomUserType(userType:string) {
```
Comment (dhDVSA) [inline src/pageobjects/des/login.pageobject.ts:77]: this is a bit misleading as the code is looking inside Super{} , but we know that there can be other users that don't live inside the super{} object . 

update naming to getRandomSuperUserType

File: src/pageobjects/des/login.pageobject.ts [hunk 7]
```diff
@@ -1,174 +1,176 @@
...
+  }
+
+  async switchToDESContext(): Promise<void> {
+    await this.waitForContextToExist(this._desAppContextTitle);
+    const DESContext = await this.getContextByTitle(this._desAppContextTitle);
+    // @ts-ignore
+    await driver.switchContext(DESContext.id);
+  }
 
-    console.log(`THIS IS VERSION  :::: ${$('.bottomright').getText()}`)
```
Comment (dhDVSA) [inline src/pageobjects/des/login.pageobject.ts:168]: Why was this deleted ?  This is quiet important

File: src/pageobjects/des/login.pageobject.ts [hunk 8]
```diff
@@ -1,174 +1,176 @@
...
+  async login(providedUser?: string): Promise<void> {
+    // pause on app launch
+    await browser.pause(2000);
```
Comment (dhDVSA) [inline src/pageobjects/des/login.pageobject.ts:99]: lets try and avoid implicit waits

**Review by dhDVSA — CHANGES_REQUESTED** [review]
some suggestions for changes

File: src/pageobjects/des/login.pageobject.ts (hunk 6 above)
Comment (Cullima) [reply src/pageobjects/des/login.pageobject.ts:77]: will update

File: src/pageobjects/des/login.pageobject.ts (hunk 7 above)
Comment (Cullima) [reply src/pageobjects/des/login.pageobject.ts:168]: why is it important? can add back in

Comment (Cullima) [reply src/pageobjects/des/login.pageobject.ts:168]: also, think this is Search code. not sure it works as is for DES?
